rust/telemetry/src/posthog.rs

```diff
@@ -0,0 +1,251 @@
+use crate::client::TelemetryClient;
+use crate::events::ProductTelemetryEvent;
+use async_trait::async_trait;
+use posthog_rs::{Client, Event};
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tracing::warn;
+
+const POSTHOG_API_KEY: &str = "phc_YeUxaojbKk5KPi8hNlx1bBKHzuZ4FDtl67kH1blv8Bh";
+const UNKNOWN_USER_ID: &str = "UNKNOWN";
+
+pub struct Posthog {
+    client: Client,
+    user_id: String,
+    is_server: bool,
+    chroma_version: String,
+    batched_events: Arc<Mutex<HashMap<String, Box<dyn ProductTelemetryEvent + Send + Sync>>>>,
+    seen_event_types: Arc<Mutex<HashSet<String>>>,
+}
+
+impl Posthog {
+    pub async fn new(user_id: Option<String>, is_server: bool, chroma_version: String) -> Self {
+        let client = posthog_rs::client(POSTHOG_API_KEY).await;
+        let user_id = user_id.unwrap_or_else(|| UNKNOWN_USER_ID.to_string());
+
+        Posthog {
+            client,
+            user_id,
+            is_server,
+            chroma_version,
+            batched_events: Arc::new(Mutex::new(HashMap::new())),
+            seen_event_types: Arc::new(Mutex::new(HashSet::new())),
+        }
+    }
+
+    async fn direct_capture(&self, event: Box<dyn ProductTelemetryEvent + Send + Sync>) {
+        let event_name = event.name();
```

Comment:
nit: in rust there is a `TryFrom` trait.

You can bound ProductTelemetryEvent: TryInto<PosthogEvent>. 

Then this method can just do .tryFrom / into